### PR TITLE
RMB-330 (MODINVSTOR-266): fix ginIndex missing lower/unaccent functions.

### PR DESCRIPTION
**Pull request by hjiebsco**
After this is released, need to bump up RMB dependencies in modules that use ginIndex: mod-inventory-storage, mod-finance-storage, mod-vendors, mod-notify, mod-audit, mod-notes, and mod-configuration.

### CODE REVIEW

File: domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java [hunk 1]
```diff
@@ -359,7 +359,7 @@ public static String convertDotPath2PostgresNotation(String prefix,
         }
         sb.append("'").append(pathParts[j]).append("'");
       }
-      if (index != null && stringType && !isFullText) {
+      if (index != null && stringType && !isFtIndex) {
```
Comment (adamdickmeiss) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java]: And you are sure it shouldn't be:
  && !isFulltext && !isFtIndex  
??

Comment (hjiebsco) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java]: That will fail the test. The purpose is to restore lower(fn_unaccent) for ginIndex. The real full text index will remain as is without those function wrappers.

File: domain-models-runtime/src/test/java/org/folio/rest/persist/ddlgen/SchemaMakerTest.java [hunk 2]
```diff
@@ -146,4 +146,27 @@ public void badVersions() throws IOException, TemplateException {
...
+  @Test
+  public void ginIndex() throws IOException, TemplateException {
+
+    SchemaMaker schemaMaker = new SchemaMaker("harvard", "circ", TenantOperation.CREATE,
+      null, null);
+
+    String json = ResourceUtil.asString("templates/db_scripts/gin_index.json");
+    schemaMaker.setSchema(ObjectMapperTool.getMapper().readValue(json, Schema.class));
+
+    System.out.println(tidy(schemaMaker.generateDDL()));
```
Comment (julianladisch) [inline domain-models-runtime/src/test/java/org/folio/rest/persist/ddlgen/SchemaMakerTest.java]: Please remove this System.out.println and this duplicate invocation of schemaMaker.generateDDL()

File: domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java (hunk 1 above)
Comment (julianladisch) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java]: Can you add a comment why the FtIndex is created without lower(f_unaccent()) but fullText and all other require it?

File: domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java [hunk 3]
```diff
@@ -323,7 +323,7 @@ private static String normalizeFieldName(String path) {
   }
 
   public static String convertDotPath2PostgresNotation(String prefix,
-    String path, boolean stringType, Index index, boolean isFullText){
+    String path, boolean stringType, Index index, boolean isFullText, boolean isFtIndex){
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java]: Can you add javadoc for this method? It should make clear why isFullText and isFtIndex can be different.

File: domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java [hunk 4]
```diff
@@ -238,7 +238,7 @@ public String generateDDL(boolean recreateIndexMode) throws IOException, Templat
           vt.setPrefix(vt.getTableName());
           Index index = indexMap.get(vt.getTableName()+"_"+normalizeFieldName(vt.getJoinOnField()));
           vt.setJoinOnField(convertDotPath2PostgresNotation(vt.getPrefix(),
-            vt.getJoinOnField(), true, index, false));
+            vt.getJoinOnField(), true, index, false, false));
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java]: unit test missing for this line

File: domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java [hunk 5]
```diff
@@ -147,7 +147,7 @@ public String generateDDL(boolean recreateIndexMode) throws IOException, Templat
             //meaning, there needs to be an index created without lowercasing / unaccenting
             //otherwise no index will be used
             ForeignKeys f = fKeys.get(j);
-            f.setFieldPath(convertDotPath2PostgresNotation("NEW",f.getFieldName(), true , null, false));
+            f.setFieldPath(convertDotPath2PostgresNotation("NEW",f.getFieldName(), true , null, false, false));
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java]: unit test missing for this line

File: domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java [hunk 6]
```diff
@@ -175,7 +175,7 @@ public String generateDDL(boolean recreateIndexMode) throws IOException, Templat
               ti.setCaseSensitive(true);
               ti.setRemoveAccents(false);
             }
-            ti.setFieldPath(convertDotPath2PostgresNotation(null,ti.getFieldName() , ti.isStringType(), ti, true));
+            ti.setFieldPath(convertDotPath2PostgresNotation(null,ti.getFieldName() , ti.isStringType(), ti, true, false));
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java]: unit test missing for this line

File: domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java [hunk 7]
```diff
@@ -197,7 +197,7 @@ public String generateDDL(boolean recreateIndexMode) throws IOException, Templat
               u.setCaseSensitive(true);
               u.setRemoveAccents(false);
             }
-            String path = convertDotPath2PostgresNotation(null,u.getFieldName(), u.isStringType(), u, false);
+            String path = convertDotPath2PostgresNotation(null,u.getFieldName(), u.isStringType(), u, false, false);
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java]: unit test missing for this line

File: domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java [hunk 8]
```diff
@@ -210,7 +210,7 @@ public String generateDDL(boolean recreateIndexMode) throws IOException, Templat
         if(ftInd != null){
           for (int j = 0; j < ftInd.size(); j++) {
             Index u = ftInd.get(j);
-            String path = convertDotPath2PostgresNotation(null,u.getFieldName(), true, u, true);
+            String path = convertDotPath2PostgresNotation(null,u.getFieldName(), true, u, true, true);
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java]: unit test missing for this line

File: domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java [hunk 9]
```diff
@@ -251,7 +251,7 @@ public String generateDDL(boolean recreateIndexMode) throws IOException, Templat
           vt.setPrefix(vt.getTableName());
           index = indexMap.get(vt.getTableName()+"_"+normalizeFieldName(vt.getJoinOnField()));
           vt.setJoinOnField( convertDotPath2PostgresNotation(vt.getPrefix(),
-            vt.getJoinOnField() , true, index, false));
+            vt.getJoinOnField() , true, index, false, false));
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java]: unit test missing for this line

File: domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java (hunk 1 above)
Comment (hjiebsco) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java]: done

File: domain-models-runtime/src/test/java/org/folio/rest/persist/ddlgen/SchemaMakerTest.java (hunk 2 above)
Comment (hjiebsco) [reply domain-models-runtime/src/test/java/org/folio/rest/persist/ddlgen/SchemaMakerTest.java]: done

**Pull request merged**